### PR TITLE
feat(core): route for add protected application

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Modify `POST /applications` for new protected app.

## Testing

Unit and integration tests.

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [x] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/routes/applications/application.ts [hunk 1]
```diff
@@ -127,12 +128,17 @@ export default function applicationRoutes<T extends AuthedRouter>(
...
     }),
     async (ctx, next) => {
-      const { oidcClientMetadata, ...rest } = ctx.guard.body;
+      const { oidcClientMetadata, protectedAppMetadata, ...rest } = ctx.guard.body;
 
       if (!EnvSet.values.isDevFeaturesEnabled && rest.type === ApplicationType.Protected) {
         throw new Error('dev feature for protected app is not enabled');
       }
 
+      assertThat(
```
Comment (simeng-li) [inline packages/core/src/routes/applications/application.ts:137]: should also guard the patch method?

File: packages/core/src/routes/applications/application.ts [hunk 2]
```diff
@@ -147,10 +153,25 @@ export default function applicationRoutes<T extends AuthedRouter>(
...
+          rest.type === ApplicationType.Protected &&
+            protectedAppMetadata && {
+              protectedAppMetadata: {
+                host: protectedAppMetadata.host,
+                origin: protectedAppMetadata.origin,
+                sessionDuration: defaultProtectedAppSessionDuration,
+                pageRules: defaultProtectedAppPageRules,
+              },
+            }
+        ),
```
Comment (simeng-li) [inline packages/core/src/routes/applications/application.ts:174]: `buildProtectedAddMetadata` ?

File: packages/core/src/routes/applications/types.ts [hunk 3]
```diff
@@ -12,8 +13,21 @@ export const applicationResponseGuard: typeof Applications.guard = EnvSet.values
...
+    protectedAppMetadata: true,
+  })
+  .extend({
+    protectedAppMetadata: z
+      .object({
+        host: z.string(),
+        origin: z.string(),
+      })
+      .optional(),
+  });
```
Comment (simeng-li) [inline packages/core/src/routes/applications/types.ts:27]: should guard by the devFeatureEnabled ?

File: packages/core/src/routes/applications/application.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/routes/applications/application.ts:140]: Updated, this field is removed from patch guard until the patch route is ready

File: packages/core/src/routes/applications/application.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/routes/applications/application.ts]: updated

File: packages/core/src/routes/applications/types.ts (hunk 3 above)
Comment (wangsijie) [reply packages/core/src/routes/applications/types.ts:38]: updated

**Review by simeng-li — APPROVED** [review]

**Pull request merged**